{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ce3bf50c-7de4-47b4-98ae-e46f37a8b87b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import cv2\n",
    "import pytesseract\n",
    "from pytesseract import Output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "eabeb53b-b6ae-4625-9e2b-0521a0c79835",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'list' object has no attribute 'split'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[0;32mIn [2]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m file_name \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mARSE.png\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m----> 2\u001b[0m gene_name \u001b[38;5;241m=\u001b[39m \u001b[43mfile_name\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msplit\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m.\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msplit\u001b[49m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m\"\u001b[39m)[\u001b[38;5;241m0\u001b[39m]\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28mprint\u001b[39m(gene_name)\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'list' object has no attribute 'split'"
     ]
    }
   ],
   "source": [
    "file_name = \"ARSE.png\"\n",
    "gene_name = file_name.split(\".\").split(\"_\")[0]\n",
    "print(gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "26a75dfc-b288-4d2b-8151-96fcda22041e",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'list' object has no attribute 'split'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[0;32mIn [3]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m file_name \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mARSE.png\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m----> 2\u001b[0m gene_name \u001b[38;5;241m=\u001b[39m \u001b[43mfile_name\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msplit\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m.\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msplit\u001b[49m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m\"\u001b[39m)[\u001b[38;5;241m0\u001b[39m]\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28mprint\u001b[39m(gene_name)\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'list' object has no attribute 'split'"
     ]
    }
   ],
   "source": [
    "file_name = \"ARSE.png\"\n",
    "gene_name = file_name.split(\".\").split(\"_\")[0]\n",
    "print(gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "88c6639d-2aaf-4d01-9980-af14b81064f2",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'gene_name' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [4]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m file_name \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mARSE.png\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m----> 2\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[43mgene_name\u001b[49m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'gene_name' is not defined"
     ]
    }
   ],
   "source": [
    "file_name = \"ARSE.png\"\n",
    "print(gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9899b6ab-65b4-487a-8495-85aef4ccda2f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ARSE.png\n"
     ]
    }
   ],
   "source": [
    "file_name = \"ARSE.png\"\n",
    "print(file_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "231ef738-8217-4ea3-b980-5446c6366f01",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A\n"
     ]
    }
   ],
   "source": [
    "print(file_name[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "de2041c4-1fe2-44f7-bfa3-7df4c7c9b78f",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_name = file_name.split(\".\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e9c0efb4-349c-4491-b816-46da6a4041f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ARSE', 'png']\n"
     ]
    }
   ],
   "source": [
    "print(gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d5f32e72-5519-4878-b862-ed2e3fa595bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_name = file_name.split(\".\")[0].split(\"_\")[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "73d2e3b3-0b3b-4352-8870-aad31d52cf7a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ARSE\n"
     ]
    }
   ],
   "source": [
    "print(gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f4a04903-20e4-4c62-bbcd-61d9de1960ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "img = cv2.imread(file_name)\n",
    "d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "keys = list(d.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9f3c0be6-87e4-493d-9b5b-d12d4d6c3896",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_boxes = len(d['text'])\n",
    "for i in range(n_boxes):\n",
    "    if gene_name in d['text'][i]:\n",
    "        (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "        img = cv2.rectangle(img, (x, y), (x + w, y + h), (0, 255, 0), 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "acbb0685-cce5-4a23-97c5-76cdb89e3eea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cv2.imwrite('img.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "b3927b9a-ff8c-45d5-95de-8a7cdca1ea01",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_name = \"KCNQ1_1.png\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "da51dde5-6a68-472b-b824-af131c08a1c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\").split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if gene_name in d['text'][i]:\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x, y), (x + w, y + h), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b76d291a-e110-4f09-b338-2bda91aeef02",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'list' object has no attribute 'split'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[0;32mIn [16]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mlabel_image\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfile_name\u001b[49m\u001b[43m)\u001b[49m\n",
      "Input \u001b[0;32mIn [15]\u001b[0m, in \u001b[0;36mlabel_image\u001b[0;34m(file_name)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mlabel_image\u001b[39m(file_name):\n\u001b[0;32m----> 2\u001b[0m     gene_name \u001b[38;5;241m=\u001b[39m \u001b[43mfile_name\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msplit\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m.\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msplit\u001b[49m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m_\u001b[39m\u001b[38;5;124m\"\u001b[39m)[\u001b[38;5;241m0\u001b[39m]\n\u001b[1;32m      4\u001b[0m     img \u001b[38;5;241m=\u001b[39m cv2\u001b[38;5;241m.\u001b[39mimread(file_name)\n\u001b[1;32m      5\u001b[0m     d \u001b[38;5;241m=\u001b[39m pytesseract\u001b[38;5;241m.\u001b[39mimage_to_data(img, output_type\u001b[38;5;241m=\u001b[39mOutput\u001b[38;5;241m.\u001b[39mDICT)\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'list' object has no attribute 'split'"
     ]
    }
   ],
   "source": [
    "label_image(file_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "fb9ff63c-d9ba-4253-8db9-3a7de8f1e9c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if gene_name in d['text'][i]:\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x, y), (x + w, y + h), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b9b7b12b-8b6a-4d81-818a-b06ddedaeb57",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image(file_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9fa519e1-34f9-4fd1-b73d-62b50c83a397",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image('CACNA1A.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "ea5a8301-214c-4eb2-ae99-93831e0f9ca0",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image('PALB2.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f23d8d5a-14a4-4300-ad37-4896209df54b",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image('LDLR.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "27140cf3-9d90-4c8d-b7e6-1bbdb2953021",
   "metadata": {},
   "outputs": [],
   "source": [
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if gene_name in d['text'][i]:\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x + 5, y + 5), (x + w + 5, y + h + 5), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6161d031-0554-4ca1-bb1b-1a92a3872805",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image('LDLR.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "73150a79-d33e-46ae-8d1a-a8afd92f79ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if gene_name in d['text'][i]:\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x - 5, y - 5), (x + w + 10, y + h + 10), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "24ed64fb-db7d-4e1b-90a5-f9dd43fb79f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image('LDLR.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "d04181a9-7ba2-4763-a158-a213144fc575",
   "metadata": {},
   "outputs": [],
   "source": [
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if gene_name in d['text'][i]:\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x - 7, y - 7), (x + w + 10, y + h + 10), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "3f5d388c-3938-4ab0-9b06-0fb0f89c461b",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image('LDLR.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "2e3e05fa-e576-4b82-980a-437b1508945c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Testing to see why nothing shows up for 'CACNA1A.png'\n",
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "        img = cv2.rectangle(img, (x - 7, y - 7), (x + w + 10, y + h + 10), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "c2643033-7d45-4bca-b366-74dcf4b8c486",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image('CACNA1A.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "f523c128-80fb-469b-86b8-26594aab21fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Testing to see why nothing shows up for 'CACNA1A.png'\n",
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        print(d['text'][i])\n",
    "        if gene_name in d['text'][i]:\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x - 7, y - 7), (x + w + 10, y + h + 10), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "af9d7ffe-9947-42eb-9579-b3b14b26b600",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n",
      "LABORATORY\n",
      "FOR\n",
      "MOLECULAR\n",
      "MEDICINE\n",
      "PARTNERS.\n",
      "CENTER\n",
      "FOR\n",
      "PERSONALIZED\n",
      "A\n",
      "teaching\n",
      "affifate\n",
      "of\n",
      "\n",
      "65\n",
      "LANDSDOWNE\n",
      "ST,\n",
      "CAMBRIDGE,\n",
      "MA02139\n",
      "E\n",
      "GENETIC\n",
      "MEDICINE\n",
      "HARVARD\n",
      "\n",
      "PHONE:\n",
      "(617)\n",
      "768-8500\n",
      "/\n",
      "FAX:\n",
      "(617)\n",
      "768-8513.\n",
      "WERETICARE\n",
      "MEDICAL\n",
      "\n",
      "http/pcpgm_partners.orgamm\n",
      "amie’\n",
      "\n",
      "\n",
      "\n",
      "Name:\n",
      "John\n",
      "Doe\n",
      "\n",
      "\n",
      "\n",
      " \n",
      " \n",
      " \n",
      "  \n",
      "\n",
      "\n",
      "\n",
      "Accession\n",
      "ID:\n",
      "0123456789\n",
      "Family\n",
      "#:\n",
      "F12345\n",
      "\n",
      "Specimen:\n",
      "Blood,\n",
      "Peripheral\n",
      "Referring\n",
      "physi\n",
      "\n",
      "Race:\n",
      "Caucasian\n",
      "Received:\n",
      "01/23/45,\n",
      "Referring\n",
      "faci\n",
      "\n",
      "\n",
      "\n",
      "GENERAL\n",
      "GENOME\n",
      "REPORT\n",
      "\n",
      "\n",
      "\n",
      "RESULT\n",
      "SUMMARY\n",
      "\n",
      "A.\n",
      "MONOGENIC\n",
      "DISEASE\n",
      "RISK:\n",
      "2\n",
      "VARIANTS\n",
      "IDENTIFIED\n",
      "\n",
      "This\n",
      "test\n",
      "identified\n",
      "2\n",
      "genetic\n",
      "variant(s)\n",
      "that\n",
      "may\n",
      "be\n",
      "responsible\n",
      "for\n",
      "existing\n",
      "disease\n",
      "or\n",
      "the\n",
      "development\n",
      "of\n",
      "disease\n",
      "in\n",
      "this\n",
      "individual's,\n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      "Disease\n",
      "Gene\n",
      "Classification\n",
      "\n",
      "Variant\n",
      "\n",
      "\n",
      "‘Al.\n",
      "Episodic\n",
      "ataxia\n",
      "type\n",
      "II\n",
      ".\n",
      "‘CACNATA.\n",
      "-\n",
      "\n",
      "(hasta\n",
      "Denso\n",
      "Poor\n",
      "coordination\n",
      "and\n",
      "balance\n",
      "Ane\n",
      "EOCtXI2\n",
      "Pathogenic\n",
      "\n",
      "(Autosomal\n",
      "Dominant)\n",
      "Progressive\n",
      "heart\n",
      "faire\n",
      "p-Thri4@AsnisX7\n",
      "Pathogenic\n",
      "\n",
      "B.\n",
      "CARRIER\n",
      "RISK:\n",
      "3\n",
      "VARIANTS\n",
      "IDENTIFIED\n",
      "\n",
      "“This\n",
      "test\n",
      "identified\n",
      "carrier\n",
      "status\n",
      "for\n",
      "3\n",
      "autosomal\n",
      "recessive\n",
      "disorder(s).\n",
      "\n",
      "\n",
      "Disease\n",
      "Phenotype\n",
      "Gene\n",
      "Classification\n",
      "\n",
      "\n",
      "Variant\n",
      "\n",
      "\n",
      "B1.\n",
      "Cystic\n",
      "fibrasis\n",
      "ae)\n",
      "C.1585-1G>A\n",
      "Pathogenio\n",
      "evidence)\n",
      "\n",
      "\n",
      "-\n",
      "CLENT\n",
      "Pathopenk\n",
      "Latent\n",
      "myotonia\n",
      "\n",
      "Bz\n",
      "Myctriacongenta\n",
      "|\n",
      "Muscledsease\n",
      "pagoex\n",
      "athogenie\n",
      "\n",
      "B3.\n",
      "Usher\n",
      "syndrome\n",
      "type\n",
      "it\n",
      "|\n",
      "Hearing\n",
      "loss\n",
      "and\n",
      "retinitis\n",
      "CenemaAgiaxt2\n",
      "Pathogenic\n",
      "None\n",
      "reported\n",
      "\n",
      "‘As\n",
      "a\n",
      "carrier\n",
      "for\n",
      "recessive\n",
      "genetic\n",
      "individuals\n",
      "at\n",
      "higher\n",
      "risk\n",
      "for\n",
      "having\n",
      "a\n",
      "child\n",
      "with\n",
      "one\n",
      "or\n",
      "more\n",
      "of\n",
      "these\n",
      "highly\n",
      "penetrant\n",
      "\n",
      "Gsarders\n",
      "To\n",
      "detomnine\n",
      "te\n",
      "rak\n",
      "forts\n",
      "ndvicuaTschlren\n",
      "tobe\n",
      "afecea,\n",
      "the\n",
      "poner\n",
      "of\n",
      "fis\n",
      "nanedeal\n",
      "would\n",
      "also\n",
      "nes\n",
      "tbe\n",
      "tsi\n",
      "or\n",
      "\n",
      "\n",
      "\n",
      "these\n",
      "variants.\n",
      "Other\n",
      "biologically\n",
      "related\n",
      "family\n",
      "members\n",
      "may\n",
      "also\n",
      "be\n",
      "carriers\n",
      "of\n",
      "these\n",
      "variants.’\n",
      "Carriers\n",
      "for\n",
      "some\n",
      "recessive\n",
      "disorders\n",
      "\n",
      "may\n",
      "be\n",
      "at\n",
      "risk\n",
      "for\n",
      "certain\n",
      "mid\n",
      "phenotypes.\n",
      "Please\n",
      "see\n",
      "variant\n",
      "descriptions\n",
      "for\n",
      "more\n",
      "information.\n",
      "\n",
      "\n",
      "\n",
      "(C.\n",
      "PHARMACOGENOMIC\n",
      "ASSOCIATIONS\n",
      "\n",
      "‘This\n",
      "test\n",
      "identified\n",
      "the\n",
      "following\n",
      "variants\n",
      "associated\n",
      "with\n",
      "drug\n",
      "use\n",
      "and\n",
      "dosing.\n",
      "Additional\n",
      "pharmacogenomic\n",
      "results\n",
      "may\n",
      "be\n",
      "requested,\n",
      "\n",
      "but\n",
      "will\n",
      "require\n",
      "adgitional\n",
      "molecular\n",
      "confirmation\n",
      "prior\n",
      "to\n",
      "disclosure.\n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      "(C1.\n",
      "Warfarin\n",
      "Decreased\n",
      "dose\n",
      "requirement.\n",
      "\n",
      "C2.\n",
      "Clopidogrel\n",
      "‘Typical\n",
      "risk\n",
      "of\n",
      "bleeding\n",
      "and\n",
      "cardiovascular\n",
      "events.\n",
      "\n",
      "C3.\n",
      "Digoxin\n",
      "Increased\n",
      "serum\n",
      "concentration\n",
      "of\n",
      "digoxin.\n",
      "\n",
      "\n",
      "(C4.\n",
      "Metformin\n",
      "‘Typical\n",
      "glycemic\n",
      "response\n",
      "to\n",
      "metformin.\n",
      "\n",
      "\n",
      "C5.\n",
      "Simvastatin\n",
      "Lower\n",
      "risk\n",
      "of\n",
      "simvastatin-elated\n",
      "myopathy.\n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      " \n",
      "\n",
      "\n",
      "\n",
      "D.\n",
      "BLOOD\n",
      "GROUPS\n",
      "\n",
      "‘This\n",
      "test\n",
      "identified\n",
      "the\n",
      "ABO\n",
      "Rh\n",
      "blood\n",
      "type\n",
      "as\n",
      "O\n",
      "positive.\n",
      "Additonal\n",
      "blood\n",
      "group\n",
      "information\n",
      "is\n",
      "available\n",
      "at\n",
      "the\n",
      "end\n",
      "of\n",
      "the\n",
      "report.\n",
      "\n",
      "\n",
      "\n",
      "It\n",
      "should\n",
      "be\n",
      "noted\n",
      "that\n",
      "the\n",
      "disease\n",
      "risk\n",
      "section\n",
      "of\n",
      "this\n",
      "report\n",
      "is\n",
      "limited\n",
      "only\n",
      "to\n",
      "variants\n",
      "with\n",
      "evidence\n",
      "for\n",
      "causing\n",
      "highly\n",
      "\n",
      "‘contributing\n",
      "to\n",
      "highly\n",
      "penetrant\n",
      "disease\n",
      "in\n",
      "a\n",
      "recessive\n",
      "manner.\n",
      "Not\n",
      "all\n",
      "variants\n",
      "identified\n",
      "have\n",
      "been.\n",
      "\n",
      "\n",
      "\n",
      "penetrant\n",
      "disease,\n",
      "or\n",
      "\n",
      "analyzed,\n",
      "and\n",
      "not\n",
      "all\n",
      "regions\n",
      "of\n",
      "the\n",
      "genome\n",
      "have\n",
      "been\n",
      "adequately\n",
      "sequenced.\n",
      "These\n",
      "results\n",
      "should\n",
      "be\n",
      "interpreted\n",
      "in\n",
      "the\n",
      "\n",
      "Py\n"
     ]
    }
   ],
   "source": [
    "label_image('CACNA1A.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "69f080ef-26fe-4af6-a80d-1948859aa77c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if gene_name in d['text'][i]:\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x - 7, y - 7), (x + w + 10, y + h + 10), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "023f3204-7b83-4f51-801b-21ff2857e5bb",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'str' object has no attribute 'keys'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[0;32mIn [59]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mlabel_image\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mCACNA1A.png\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "Input \u001b[0;32mIn [56]\u001b[0m, in \u001b[0;36mlabel_image\u001b[0;34m(file_name)\u001b[0m\n\u001b[1;32m      4\u001b[0m img \u001b[38;5;241m=\u001b[39m cv2\u001b[38;5;241m.\u001b[39mimread(file_name)\n\u001b[1;32m      5\u001b[0m d \u001b[38;5;241m=\u001b[39m pytesseract\u001b[38;5;241m.\u001b[39mimage_to_data(img)\n\u001b[0;32m----> 6\u001b[0m keys \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(\u001b[43md\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkeys\u001b[49m())\n\u001b[1;32m      8\u001b[0m n_boxes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(d[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtext\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[1;32m      9\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(n_boxes):\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'str' object has no attribute 'keys'"
     ]
    }
   ],
   "source": [
    "label_image('CACNA1A.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "013f32b3-6fb4-4f4e-9cb8-27730f962d3d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_similarity(a, b):\n",
    "    chars_to_remove = \"1ltLT\"\n",
    "    for char in chars_to_remove:\n",
    "        a = a.replace(char, \"\")\n",
    "        b = b.replace(char, \"\")\n",
    "    if (a in b):\n",
    "        return true\n",
    "    return false\n",
    "\n",
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if check_similarity(gene_name, d['text'][i]):\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x - 7, y - 7), (x + w + 10, y + h + 10), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "bf372646-272e-403b-af16-b66953131034",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'str' object has no attribute 'keys'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[0;32mIn [61]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mlabel_image\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mCACNA1A.png\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "Input \u001b[0;32mIn [60]\u001b[0m, in \u001b[0;36mlabel_image\u001b[0;34m(file_name)\u001b[0m\n\u001b[1;32m     13\u001b[0m img \u001b[38;5;241m=\u001b[39m cv2\u001b[38;5;241m.\u001b[39mimread(file_name)\n\u001b[1;32m     14\u001b[0m d \u001b[38;5;241m=\u001b[39m pytesseract\u001b[38;5;241m.\u001b[39mimage_to_data(img)\n\u001b[0;32m---> 15\u001b[0m keys \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(\u001b[43md\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkeys\u001b[49m())\n\u001b[1;32m     17\u001b[0m n_boxes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(d[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtext\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[1;32m     18\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(n_boxes):\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'str' object has no attribute 'keys'"
     ]
    }
   ],
   "source": [
    "label_image('CACNA1A.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "46a22495-1e12-43f3-98de-8c0b90f2bf3e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_similarity(a, b):\n",
    "    chars_to_remove = \"1ltLT\"\n",
    "    for char in chars_to_remove:\n",
    "        a = a.replace(char, \"\")\n",
    "        b = b.replace(char, \"\")\n",
    "    if (a in b):\n",
    "        return true\n",
    "    return false\n",
    "\n",
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if check_similarity(gene_name, d['text'][i]):\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x - 7, y - 7), (x + w + 10, y + h + 10), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "f72d7d58-ac1e-4d34-a33e-90e1298f3bbc",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'false' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [63]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mlabel_image\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mCACNA1A.png\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "Input \u001b[0;32mIn [62]\u001b[0m, in \u001b[0;36mlabel_image\u001b[0;34m(file_name)\u001b[0m\n\u001b[1;32m     17\u001b[0m n_boxes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(d[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtext\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[1;32m     18\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(n_boxes):\n\u001b[0;32m---> 19\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[43mcheck_similarity\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgene_name\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43md\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mtext\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m:\n\u001b[1;32m     20\u001b[0m         (x, y, w, h) \u001b[38;5;241m=\u001b[39m (d[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mleft\u001b[39m\u001b[38;5;124m'\u001b[39m][i], d[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtop\u001b[39m\u001b[38;5;124m'\u001b[39m][i], d[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mwidth\u001b[39m\u001b[38;5;124m'\u001b[39m][i], d[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mheight\u001b[39m\u001b[38;5;124m'\u001b[39m][i])\n\u001b[1;32m     21\u001b[0m         img \u001b[38;5;241m=\u001b[39m cv2\u001b[38;5;241m.\u001b[39mrectangle(img, (x \u001b[38;5;241m-\u001b[39m \u001b[38;5;241m7\u001b[39m, y \u001b[38;5;241m-\u001b[39m \u001b[38;5;241m7\u001b[39m), (x \u001b[38;5;241m+\u001b[39m w \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m10\u001b[39m, y \u001b[38;5;241m+\u001b[39m h \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m10\u001b[39m), (\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m255\u001b[39m, \u001b[38;5;241m0\u001b[39m), \u001b[38;5;241m2\u001b[39m)\n",
      "Input \u001b[0;32mIn [62]\u001b[0m, in \u001b[0;36mcheck_similarity\u001b[0;34m(a, b)\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (a \u001b[38;5;129;01min\u001b[39;00m b):\n\u001b[1;32m      7\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m true\n\u001b[0;32m----> 8\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfalse\u001b[49m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'false' is not defined"
     ]
    }
   ],
   "source": [
    "label_image('CACNA1A.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "0efbd789-a5ae-413b-8b3b-3a70e979c591",
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_similarity(a, b):\n",
    "    chars_to_remove = \"1ltLT\"\n",
    "    for char in chars_to_remove:\n",
    "        a = a.replace(char, \"\")\n",
    "        b = b.replace(char, \"\")\n",
    "    if (a in b):\n",
    "        return True\n",
    "    return False\n",
    "\n",
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if check_similarity(gene_name, d['text'][i]):\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x - 7, y - 7), (x + w + 10, y + h + 10), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "6647841c-0380-4b70-a118-6c239e633d6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image('CACNA1A.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "6741ab6d-2f4c-418d-86fb-c2833dbf0da8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_similarity(a, b):\n",
    "    chars_to_remove = \"1ltLT\"\n",
    "    for char in chars_to_remove:\n",
    "        a = a.replace(char, \"\")\n",
    "        b = b.replace(char, \"\")\n",
    "    if (a in b):\n",
    "        return True\n",
    "    return False\n",
    "\n",
    "def label_image(file_name):\n",
    "    gene_name = file_name.split(\".\")[0].split(\"_\")[0]\n",
    "    \n",
    "    img = cv2.imread(file_name)\n",
    "    d = pytesseract.image_to_data(img, output_type=Output.DICT)\n",
    "    keys = list(d.keys())\n",
    "    \n",
    "    n_boxes = len(d['text'])\n",
    "    for i in range(n_boxes):\n",
    "        if check_similarity(gene_name, d['text'][i]):\n",
    "            (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])\n",
    "            img = cv2.rectangle(img, (x - 6, y - 6), (x + w + 9, y + h + 9), (0, 255, 0), 2)\n",
    "            \n",
    "    cv2.imwrite('output.png', img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "9d15b26e-7b9e-44b0-8944-4d4786765c87",
   "metadata": {},
   "outputs": [],
   "source": [
    "label_image('CACNA1A.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2aa1bd77-a9d0-40e6-8053-534b57966524",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
